assets/src/web-stories-block/block/api/useStoryApi.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo, useReducer } from 'react';
+import queryString from 'query-string';
+
+/**
+ * Internal dependencies
+ */
+import {
+  STORY_STATUSES,
+  STORY_SORT_OPTIONS,
+  ORDER_BY_SORT,
+  STORIES_PER_REQUEST,
+  STORY_STATUS,
+} from '../../../dashboard/constants';
+import storyReducer, {
+  defaultStoriesState,
+  ACTION_TYPES as STORY_ACTION_TYPES,
+} from '../../../dashboard/app/reducer/stories';
+import { ERRORS } from '../../../dashboard/app/textContent';
+
+const useStoryApi = (dataAdapter, { editStoryURL, storyApi }) => {
+  const [state, dispatch] = useReducer(storyReducer, defaultStoriesState);
```

Comment:
@swissspidy 
As we discussed on PR [#150 ](https://github.com/rtCamp/web-stories-wp/pull/150#discussion_r544321911), in this PR I've taken some part from the web stories dashboard and created separate providers for the block.

Still for the stories data, I'm relying on the existing `storyReducer`.

Reply:
> Still for the stories data, I'm relying on the existing `storyReducer`.

Because it's so big & complex?

Reply:
It does look big and complex at first but got hang of it 😅 . Now for the why we are still using it, we have added feature to rearrange selected stories, which relies on the data return from the existing storyReducer.

Now we can decouple it but at the moment it could take more time to go through the components used and identify how to remove the complete reliance.

This PR has removed reliance on dashboard API all but for the storyReducer. let me know your thoughts.